### DISCUSSION TITLE
Mac VM (guest) + Linux (host) + BlackHole = route audio to Linux

### DISCUSSION THREAD

**Discussion by avently** [mikebrady/shairport-sync#1913] [Q&A]
Hello. Thank you for your hard work. 
I'm trying to play audio from Mac (running in Qemu virtual machine) by using virtual output audio channel made by [BlackHole](https://github.com/ExistentialAudio/BlackHole). The VM is running from Linux host. The idea is to route that virtual audio channel to local Linux host. Guest Mac is accessible via network from host so I can setup SSH connection with port forwarding if needed.

For now when I run shairport-sync with default config I don't see an option to share sound to host via Airplay menu on Mac (top right corner in menu). However I see that there is an option to do it from Music app on Mac but it fails to connect to the host when I choose the host name in that Music app. 

I don't have any firewalls so everything should just work if I'm able to preperly setup both sides (guest and host).

Can you suggest me a configuration that will achive the idea?

**Comment by mikebrady** [comment]
Thanks for this interesting question. There are many unknowns in it, so I think the approach would be to check each piece separately as much as possible.

If you set Shairport Sync's `log_verbosity` to `1` and enable the gathering of statistics, you should be able to see if it is receiving and attempting to play audio.

VMs typically don't do real-time timing very well, and this may be stopping synchronisation from working and thus causing the output to remain silent. To avoid this, it might we worth using a backend that doesn't attempt to do active synchronisation, so a backend like the `stdout` or the unix `pipe` backend might be preferable to the `alsa` backend.

The "Discovery" apps on Mac and iOS should help to see if Shairport Sync is correctly advertised and visible in Bonjour on the network, as will `avahi-browse` on the Linux device.
